### PR TITLE
Add recycle bin for capture central

### PR DESCRIPTION
**Pull request by clemencetbk**
Add recycle bin with the option to restore deleted items in Capture Central (option to delete permanently has not yet been added)

### CODE REVIEW

File: applications/d2l-capture-central/src/components/capture-central-list.js [hunk 1]
```diff
@@ -173,7 +175,7 @@ export class CaptureCentralList extends DependencyRequester(InternalLocalizeMixi
 			rootStore.routingStore,
 			'queryParams',
 			change => {
-				if (this.loading || rootStore.routingStore.page !== this.page) {
+				if (this.loading || `/${  rootStore.routingStore.page}` !== this.page) {
```
Comment (MichaelKovalchuk) [inline applications/d2l-capture-central/src/components/capture-central-list.js:178]: spacing `/${  `

File: applications/d2l-capture-central/src/components/recycle-bin/recycle-bin-list.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+			id=${item.id}
+			revision-id=${item.revisionId}
+			title=${item.title}
+			@recycle-bin-item-restored=${this.recycleBinItemRestoredHandler}
+		>
+			<d2l-icon icon="tier1:file-video" slot="icon"></d2l-icon>
+			<div slot="title" class="title">${item.title}</div>
+			<div slot="type">${item.type}</div>
+			<relative-date slot="date" value=${item[this.dateField]}></relative-date>
+			<div slot="expiry" class=d2l-body-small>Permanently deleted in ${numDays} ${numDays === 1 ? 'day' : 'days'}</div>
```
Comment (MichaelKovalchuk) [inline applications/d2l-capture-central/src/components/recycle-bin/recycle-bin-list.js:116]: translation is missing?

File: applications/d2l-capture-central/src/components/recycle-bin/recycle-bin-item.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+					<div slot="date">
+						<slot name="date"></slot>
+						<slot name="expiry"></slot>
+					</div>
+				</recycle-bin-list-columns>
+
+				<div slot="actions" id="actions" class="actions">
+					<d2l-button
+						class="d2l-capture-central-restore-button"
+						@click="${this.restore}"
```
Comment (MichaelKovalchuk) [inline applications/d2l-capture-central/src/components/recycle-bin/recycle-bin-item.js:90]: you are also planning to add delete-button here?
Can we have a confirmation dialog before the item is permanently deleted?

File: applications/d2l-capture-central/src/components/recycle-bin/recycle-bin-item.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+			.catch(() => downloadWindow.close());
+	}
+	async rename(newTitle) {
+		if (this.title !== newTitle) {
+			this.dispatchRenameEvent(newTitle);
+
+			if (!this.content) {
+				this.content = await this.apiClient.getContent(this.id);
+			}
+			const updatedContent = Object.assign({}, this.content, { title: newTitle });
```
Comment (MichaelKovalchuk) [inline applications/d2l-capture-central/src/components/recycle-bin/recycle-bin-item.js:153]: maybe?
```js
await this.apiClient.updateContent({
				id: this.id,
				body: {
                   ...this.content,
                   title: newTitle,
                },
			});
```

File: applications/d2l-capture-central/src/components/recycle-bin/recycle-bin-item.js [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+	}
+	download() {
+		// Safari blocks calls to window.open within an async call
+		const downloadWindow = window.open('', '_blank');
+		this.apiClient.getSignedUrlForRevision({
+			contentId: this.id,
+			revisionId: this.revisionId
+		})
+			.then(res => {
+				if (res && res.value) {
```
Comment (MichaelKovalchuk) [inline applications/d2l-capture-central/src/components/recycle-bin/recycle-bin-item.js:138]: nit: `res?.value`

File: applications/d2l-capture-central/src/components/recycle-bin/recycle-bin-item.js [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+			});
+		}
+	}
+	restore() {
+		this.dispatchRestoreEvent();
+	}
+
+	titleInputChangedHandler() {
+		const titleInputElement = this.shadowRoot.querySelector('#rename-input');
+		const titleInputValue = titleInputElement && titleInputElement.value;
```
Comment (MichaelKovalchuk) [inline applications/d2l-capture-central/src/components/recycle-bin/recycle-bin-item.js:166]: nit: `titleInputElement?.value`

**Review by MichaelKovalchuk — APPROVED** [review]

File: applications/d2l-capture-central/src/components/recycle-bin/recycle-bin-item.js (hunk 3 above)
Comment (clemencetbk) [reply applications/d2l-capture-central/src/components/recycle-bin/recycle-bin-item.js:90]: Yes, I can add that with the delete button in a different PR

File: applications/d2l-capture-central/src/components/recycle-bin/recycle-bin-item.js (hunk 5 above)
Comment (clemencetbk) [reply applications/d2l-capture-central/src/components/recycle-bin/recycle-bin-item.js:138]: Actually this is something I accidentally copied from content-list-item.js... I don't think you need to be able to download deleted items so I think I'll remove this function altogether.

Comment (MichaelKovalchuk) [reply applications/d2l-capture-central/src/components/recycle-bin/recycle-bin-item.js:138]: same for the title change?

**Review by MichaelKovalchuk — APPROVED** [review]

**Pull request merged**